### PR TITLE
Emit 10.0.2xxxx for corresponding Windows builds

### LINKED ISSUE #11: Distinguish between Windows 10 and 11
Right now Windows 11 gets lumped in with Windows 10. This is to do with how Microsoft decided to distinguish between them (which is to say: hardly at all).

I _think_ all we need to do is to augment [these lines](https://github.com/mozilla/firefox-public-data-report-etl/blob/main/public_data_report/hardware_report/hardware_report.py#L53-L57) with a call to `mozfun.norm.windows_version_info`.

Actually, `windows_version_info` returns stuff like "Windows 11" which won't fit in well. May need our own UDF. But maybe that's all we need is to return `11` from there and it'll percolate through? (I don't know)

### PR DESCRIPTION
**Pull request by saschanaz**
Then the transposer can rename that into Windows 11 [here](https://github.com/mozilla/ensemble-transposer/blob/3a28e5da3f7f1211b8e96ae609ec68bbb87765df/config/datasets/hardware.json#L372-L375).

Fixes #11.

### CODE REVIEW

**Comment by saschanaz** [comment]
Turns out that:

1. All Windows 10.0.1xxxx is Windows 10 (starting from 10240, although there was pre-10000 preview builds, to the latest 19045)
2. All Windows 10.0.2xxxx is Windows 11 (starting from 22000, dev builds are 23xxx, canaries are 25xxx)

So I opted to trim the digits below the first one to group them together.

**Comment by whd** [comment]
I've requested reviews from the owners of the DAG that I believe uses this code: https://github.com/mozilla/telemetry-airflow/blob/main/dags/firefox_public_data_report.py#L28. @akkomar is PTO so unless this is urgent I wouldn't expect a response until he's back. I suspect that DE doesn't actively monitor this repo so if you want to make sure this work is on their radar file a ticket via one of the mechanisms described on https://mozilla-hub.atlassian.net/wiki/spaces/DATA/overview?homepageId=6848546 or ask about it in #data-help in slack.

**Comment by saschanaz** [comment]
ᅟSo nobody triages the issues here? That's very sad.

**Comment by saschanaz** [comment]
Anyway, https://bugzilla.mozilla.org/show_bug.cgi?id=1868663 if I understood the page right.

**Comment by whd** [comment]
> ᅟSo nobody triages the issues here? That's very sad and needs to be fixed.

That's correct. I was asked by @jrmuizel via Slack to look at this which is why I'm responding.

You can read about some of the background for this service in https://mozilla-hub.atlassian.net/browse/DSRE-45?focusedCommentId=627600 and https://mozilla-hub.atlassian.net/browse/DSRE-987?focusedCommentId=584890. I suggest speaking with @rafrombrc if you're interested in improving the situation. Beyond the work done in https://mozilla-hub.atlassian.net/browse/DSRE-285 / https://mozilla-hub.atlassian.net/browse/DENG-1024 to migrate off of some legacy infrastructure backing the service, any improvements to the public data report are AIUI very low on the DE priority list.

**Review by akkomar — APPROVED** [review]

File: public_data_report/hardware_report/hardware_report.py [hunk 1]
```diff
@@ -50,9 +50,16 @@ def load_data(spark, date_from, date_to):
...
-                CONCAT(REGEXP_EXTRACT(environment.system.os.version, r"^[0-9]+"), '.x'),
-                environment.system.os.version),
+            CASE
+                WHEN environment.system.os.name IN ('Linux', 'Darwin')
+                    THEN CONCAT(REGEXP_EXTRACT(environment.system.os.version, r"^[0-9]+"), '.x')
+                WHEN
+                    environment.system.os.name = 'Windows_NT' and
+                    environment.system.os.version = '10.0' and
+                    environment.system.os.windows_build_number >= 10000
+                    THEN CONCAT('10.0.', CAST(CAST(environment.system.os.windows_build_number / 10000 AS INT64) AS STRING), 'xxxx')
```
Comment (akkomar) [inline public_data_report/hardware_report/hardware_report.py:60]:
```suggestion
                    THEN CONCAT(
                        '10.0.',
                        CAST(CAST(
                            environment.system.os.windows_build_number / 10000 AS INT64
                            ) AS STRING), 'xxxx'
                        )
```

**Pull request merged**